src/2025h2/parallel-front-end.md

```diff
@@ -0,0 +1,75 @@
+# Promoting Parallel Front End
+
+| Metadata         |                                                   |
+| :--------------- | ------------------------------------------------- |
+| Point of contact | @SparrowLii                                       |
+| Teams            | <!-- TEAMS WITH ASKS -->                          |
+| Task owners      | <!-- TASK OWNERS -->                              |
+| Status           | Proposed                                          |
+| Tracking issue   | [https://github.com/rust-lang/rust/issues/113349] |
+| Zulip channel    | [#t-compiler/wg-parallel-rustc][channel]          |
+
+[channel]: https://rust-lang.zulipchat.com/#narrow/channel/187679-t-compiler.2Fwg-parallel-rustc/
+
+
+## Summary
+
+Continue to parallelize front-end stabilization and performance improvements, continuing from the [2025h1 goal](https://rust-lang.github.io/rust-project-goals/2025h1/parallel-front-end.html).
+
+## Motivation
+
+Verify and resolve the few remaining deadlock and other issues, stabilize the features, and try to further improve the parallel compilation performance through several means.
+
+### The status quo
+
+Parallel front end progressed well in the first half of the year. We resolved the deadlock issue caused by the deadlock handler, added the rustc-rayon  that the parallel front end depends on to the rustc working tree, and optimized the work-stealing scheduling algorithm to avoid deadlocks in the parallelization of rustc queries.
```

Comment:
```suggestion
Parallel front end progressed well in the first half of the year. We resolved the deadlock issue caused by the deadlock handler, added the rustc-rayon dependency of the parallel front end to the rustc working tree, and changed the work-stealing scheduling algorithm to avoid deadlocks in parallel execution of rustc queries.
```